idx/memory/memory_test.go

```diff
@@ -279,6 +279,30 @@ func TestDelete(t *testing.T) {
 	})
 }
 
+func TestDeleteWithLotsOfSeries(t *testing.T) {
```

Comment:
my point is that the purpose of unit tests is testing functionality. if you want to verify performance criteria (such as x must take no more than y time) then that is what benchmarks are for.